src/main/java/org/fenixedu/academic/service/StudentWarningsDefaultCheckers.java

```diff
@@ -0,0 +1,40 @@
+package org.fenixedu.academic.service;
+
+import org.fenixedu.academic.domain.IdentificationDocumentExtraDigit;
+import org.fenixedu.academic.domain.IdentificationDocumentSeriesNumber;
+import org.fenixedu.academic.domain.Person;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.domain.person.IDDocumentType;
+import org.fenixedu.academic.domain.student.Student;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.function.Function;
+
+public class StudentWarningsDefaultCheckers {
+
+    public final static Function<Student, Collection<String>> WARNING_VALID_ID_DOCUMENT = student -> {
+        Collection<String> warnings = new ArrayList<>();
+        Person person = student.getPerson();
+        if (person.getIdDocumentType().equals(IDDocumentType.IDENTITY_CARD)) {
```

Comment:
We can do the following instead, because enums are the same instance in the JVM
`person.getIdDocumentType() == IDDocumentType.IDENTITY_CARD`
It will have a slightly better performance, and this code will be called allot.